src/PlayerRenderComponent.cpp

```diff
@@ -0,0 +1,44 @@
+#include "PlayerRenderComponent.h"
+#include "GameObject.h"
+
+PlayerRenderComponent::PlayerRenderComponent(std::shared_ptr<Shape> shape,
+   std::shared_ptr<Program> shaderProgram, std::shared_ptr<Material> material)
+   : RenderComponent(shape, shaderProgram, material) {
+
+}
+
+PlayerRenderComponent::~PlayerRenderComponent() {}
+
+void PlayerRenderComponent::draw(std::shared_ptr<MatrixStack> P,
+   std::shared_ptr<MatrixStack> M, std::shared_ptr<MatrixStack> V) {
+   ShaderManager& shaderManager = ShaderManager::instance();
+   shaderManager.bindShader(shaderProgram_->name);
+
+	// Bind perspective and view tranforms
+	glUniformMatrix4fv(shaderProgram_->getUniform("P"), 1, GL_FALSE, glm::value_ptr(P->topMatrix()));
+	glUniformMatrix4fv(shaderProgram_->getUniform("V"), 1, GL_FALSE, glm::value_ptr(V->topMatrix()));
+
+	// Bind light properties
+	glUniform3f(shaderProgram_->getUniform("lightPos"), curLight.x, curLight.y, curLight.z);
+	glUniform3f(shaderProgram_->getUniform("lightClr"), curLight.r, curLight.g, curLight.b);
+
+	// Bind material properties
+	glUniform3f(shaderProgram_->getUniform("MatAmb"), material_->rAmb, material_->gAmb, material_->bAmb);
+	glUniform3f(shaderProgram_->getUniform("MatDif"), material_->rDif, material_->gDif, material_->bDif);
+	glUniform3f(shaderProgram_->getUniform("MatSpc"), material_->rSpc, material_->gSpc, material_->bSpc);
+	glUniform1f(shaderProgram_->getUniform("MatShiny"), material_->shininess);
+
+	// Set up and bind model transform
+	M->pushMatrix();
+	M->loadIdentity();
+
+	M->multMatrix(holder_->transform.getTransform());
```

Comment:
Yea, I did appreciate the abstraction to be honest. Although I would say we should extend the `MatrixTransform` class to support multiple rotates, transform orders, etc.